gui/jails/forms.py

```diff
@@ -405,12 +416,28 @@ def save(self):
                     jail_flags |= WARDEN_SET_FLAGS_FLAGS
                     jail_set_args['jflags'] = val
 
-                jail_set_args['flags'] = jail_flags
-                try:
-                    w.set(**jail_set_args)
-                except Exception as e:
-                    self.errors['__all__'] = self.error_class([_(e.message)])
-                    return
+            else:
+                if key == 'jail_mac':
+                    duplicate_mac = 0
+                    while True:
+                        mac_address = generate_randomMAC()
+                        jail_list = Jails.objects.all()
```

Comment:
Why query this list on every iteration of the while loop? you can just do it once outside the while loop and store it in `jail_list` variable